wfp/mmp_timestamp_correction.py

```diff
@@ -0,0 +1,201 @@
+			e_file[index:index+4] = new_vehicle_end_time_bytes
+
+		sensor_end_time = unpack('>i', e_file[index+4:index+8])[0]
+		sensor_end_time = (d + timedelta(seconds=sensor_end_time))
+		if sensor_end_time < datetime(2018,1,1):
+			new_sensor_end_time = sensor_end_time + relativedelta(years=80)
+			new_sensor_end_time_seconds = int((new_sensor_end_time - epoch).total_seconds())
+			new_sensor_end_time_bytes = pack('>i', new_sensor_end_time_seconds)
+			e_file[index+4:index+8] = new_sensor_end_time_bytes
+
+	write_to_e = open(file, 'wb')
+	write_to_e.write(e_file)
+	write_to_e.close()
+
+
+#-----------------------------------------------------------------------------------------
+def process_m_file(directory, profile_count):
+	m_file_name = 'M'+str(profile_count).rjust(7,'0')+'.DAT'
+	file = os.path.join(directory, m_file_name)
+	try: m_file = bytearray(open(file, 'rb').read())
+	except: return
+	d = datetime.strptime("01-01-1970", "%m-%d-%Y")
+
+	package_size = unpack('>H', m_file[0:2])[0]
+	index = 2
+	while len(m_file)-index > 64:
+		time_stamp = unpack('>i', m_file[index:index+4])[0]
+		time_stamp = (d + timedelta(seconds=time_stamp))
+		if time_stamp < datetime(2018,1,1):
+			new_time_stamp = time_stamp + relativedelta(years=80)
+			new_time_stamp_seconds = int((new_time_stamp - epoch).total_seconds())
+			new_time_stamp_bytes = pack('>i', new_time_stamp_seconds)
+			m_file[index:index+4] = new_time_stamp_bytes
+		index += package_size
+	index+=package_size
+
+	if len(m_file) > index:
+		motion_start_time = unpack('>i', m_file[index:index+4])[0]
+		motion_start_time = (d + timedelta(seconds=motion_start_time))
+		if motion_start_time < datetime(2018,1,1):
+			new_motion_start_time = motion_start_time + relativedelta(years=80)
+			new_motion_start_time_seconds = int((new_motion_start_time - epoch).total_seconds())
+			new_motion_start_time_bytes = pack('>i', new_motion_start_time_seconds)
+			m_file[index:index+4] = new_motion_start_time_bytes
+
+		motion_end_time = unpack('>i', m_file[index+4:index+8])[0]
+		motion_end_time = (d + timedelta(seconds=motion_end_time))
+		if motion_end_time < datetime(2018,1,1):
+			new_motion_end_time = motion_end_time + relativedelta(years=80)
+			new_motion_end_time_seconds = int((new_motion_end_time - epoch).total_seconds())
+			new_motion_end_time_bytes = pack('>i', new_motion_end_time_seconds)
+			m_file[index+4:index+8] = new_motion_end_time_bytes
+
+	write_to_m = open(file, 'wb')
+	write_to_m.write(m_file)
+	write_to_m.close()
+
+
+directory = ''
+os.chdir(directory)
+file_list = []
+for file in glob.glob('?*[0-9].DAT'):
+	file_list.append(int(re.findall('.(\d+).DAT', file)[0]))
+total_profile = max(file_list)
+
+
+for profile_count in range(0, total_profile+1):
+	print(profile_count)
+	process_a_file(directory, profile_count)
+	try: process_c_file(directory, profile_count)
+	except: pass
+	process_e_file(directory, profile_count)
+	process_m_file(directory, profile_count)
+
```

Comment:
Tested on my machine and it worked well. Would recommend changing "MMP" to "WFP" in the directory structure for the code. Would also recommend adding shebang and python path to top of script and adding more documentation (beefing up the script description and adding function documentation)